drivers/flash/soc_flash_mcux.c

```diff
@@ -85,13 +85,26 @@ static int flash_mcux_write_protection(struct device *dev, bool enable)
 	return -EIO;
 }
 
+#if defined(CONFIG_FLASH_PAGE_LAYOUT)
+static void flash_not_implemented(void)
+{
+	__ASSERT(0,
```

Comment:
k_panic() sounds like a better fit

Reply:
Nobody else really uses k_panic() though, so I think the original assert was better.